lib/generator.test.js

```diff
@@ -113,15 +126,154 @@ test("overwrite file if already exist; on overwrite = true", async() => {
   })
 })
 
-test("throws error if target already exist on", async() => {
+test("throws error if target directory already exist", async() => {
   expect.assertions(1)
 
-  return generator("/template", "test-app", {
+  try {
+    await generator("/template", "test-app", {
+      variables: {
+        name: "sinix",
+        version: "0.1.0",
+      },
+    })
+  } catch (err){
+    expect(err.code).toBe("EEXIST")
+  }
+})
+
+test("don't add file if ignored", async() => {
+  expect.assertions(1)
+
+  const s = generator("/template", "app", {
+    overwrite: true,
+    variables: {
+      name: "sinix",
+      version: "0.1.0",
+    },
+  })
+
+  s.ignore("src/main.js")
+
+  await s
+
+  expect(fs.existsSync("app/src/main.js")).toBe(false)
+})
+
+test("copy file to a custom place", async() => {
+  expect.assertions(1)
+
+  const s = generator("/template", "app", {
+    variables: {
+      name: "sinix",
+      version: "0.1.0",
+    },
+  })
+
+  s.add("src/main.js", "src/bleh.js")
+
+  await s
+
+  expect(fs.existsSync("app/src/bleh.js")).toBe(true)
```

Comment:
Also check if the file is not copied to the default path
```suggestion
  expect(fs.existsSync("app/src/bleh.js")).toBe(true)
  expect(fs.existsSync("app/src/main.js")).toBe(false)
```